app/src/main/java/com/innowisegroup/datetimepicker/FragmentDatePicker.kt

```diff
@@ -62,88 +50,101 @@ class FragmentDatePicker : Fragment() {
             )
         }
         day?.wrapSelectorWheel = wrapSelectionWheel
-        day?.setOnValueChangedListener { _: NumberPicker?, _: Int, newVal: Int ->
+        day?.setOnValueChangedListener { _, _, newVal: Int ->
             refreshDateValue(
-                    localDate?.withDayOfMonth(newVal)!!
+                localDate?.withDayOfMonth(newVal)!!
             )
         }
 
         month?.minValue = MONTH_MIN_VALUE
         month?.maxValue = MONTH_MAX_VALUE
         month?.value = localDate?.getMonth()!!
         month?.setDividerColor(
-                month,
-                ResourcesCompat.getDrawable(resources, R.drawable.number_picker_divider_color, null)
+            month,
+            ResourcesCompat.getDrawable(resources, R.drawable.number_picker_divider_color, null)
         )
         month?.setFormatter { i: Int ->
             String.format(
-                    Locale.getDefault(),
-                    "%02d",
-                    i
+                Locale.getDefault(),
+                "%02d",
+                i
             )
         }
         month?.wrapSelectorWheel = wrapSelectionWheel
-        month?.setOnValueChangedListener { _: NumberPicker?, _: Int, newVal: Int ->
+        month?.setOnValueChangedListener { _, _, newVal: Int ->
             refreshDateValue(
-                    localDate?.withMonth(newVal)!!
+                localDate?.withMonth(newVal)!!
             )
         }
 
         year?.minValue = minLocalDate?.getMinYear()!!
         year?.maxValue = maxLocalDate?.getMaxYear()!!
         year?.value = localDate?.getYear()!!
         year?.setDividerColor(
-                year,
-                ResourcesCompat.getDrawable(resources, R.drawable.number_picker_divider_color, null)
+            year,
+            ResourcesCompat.getDrawable(resources, R.drawable.number_picker_divider_color, null)
         )
         year?.setFormatter { i: Int ->
             String.format(
-                    Locale.getDefault(),
-                    "%04d",
-                    i
+                Locale.getDefault(),
+                "%04d",
+                i
             )
         }
         year?.wrapSelectorWheel = wrapSelectionWheel
-        year?.setOnValueChangedListener { _: NumberPicker?, _: Int, newVal: Int ->
+        year?.setOnValueChangedListener { _, _, newVal: Int ->
             refreshDateValue(
-                    localDate?.withYear(newVal)!!
+                localDate?.withYear(newVal)!!
             )
         }
-        refreshDateValue(
-                LocalDate().of(
-                        localDate?.getDay()!!,
-                        localDate?.getMonth()!!,
-                        localDate?.getYear()!!
-                )
-        )
         return view
     }
 
     private fun refreshDateValue(newValue: LocalDate) {
-        if (day == null) {
-            return
-        }
-        localDate = newValue
-        day?.maxValue = localDate?.lengthOfMonth()!!
```

Comment:
Why did you delete these lines? Now you have the incorrect value in the tab. These lines helps to set the correct value to the tab and set how many days in a month. Return these lines